GitUI/Infrastructure/Telemetry/DiagnosticsClient.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Windows.Forms;
+using GitExtUtils.GitUI;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace GitUI.Infrastructure.Telemetry
+{
+    // The original idea and the implementation are borrowed from  https://github.com/NuGetPackageExplorer/NuGetPackageExplorer
+    // Credits to Oren Novotny
+
+    public static class DiagnosticsClient
+    {
+        private static bool _initialized;
+
+        private static TelemetryClient _client;
+
+        public static void Initialize(bool isDirty)
+        {
+            TelemetryConfiguration.Active.InstrumentationKey = "2ef275e3-8850-4305-9d7c-825a60c3d296";
+            TelemetryConfiguration.Active.TelemetryChannel.DeveloperMode = Debugger.IsAttached;
```

Comment:
You no longer need this if you use the latest AppInsights.WindowsDesktop library